stable/mongodb/templates/statefulset-primary-rs.yaml

```diff
@@ -111,7 +111,7 @@ spec:
             value: {{ .Values.mongodbDatabase | quote }}
           {{- end }}
             {{- if .Values.usePassword }}
-            {{- if and .Values.mongodbUsername (or .Values.mongodbPassword .Values.existingSecret) }}
+            {{- if and .Values.mongodbUsername .Values.mongodbDatabase }}
```

Comment:
I used the same logic than in the already existing secret file : if the user provide mongodbUsername and a mongodbDatabase the password must be provided to.
You can see on line 118 of the file that it will not affect existing secrets support because if an existing secret is provided the password is read from it.

I just want the chart to generate a secret for me if not provided. The logic was already implemented in the secrets.yaml file but it didn't work because of the line 114 of statefulset-primary-rs.yaml.

Reply:
You're right, it doesn't affect existing secrets. 

Regarding generating random passwords when not provided, I don't find it useful since it's a password for a custom user/database (so I'm supposed to know what password I want to use). However, I think it's makes no harm to support that feature.